python-frontend/src/main/java/org/sonar/python/types/v2/TypeCheckBuilder.java

```diff
@@ -227,5 +232,40 @@ private static Set<PythonType> collectTypes(PythonType type) {
     }
   }
 
+  record IsIdentityComparableWith(PythonType expectedType) implements TypePredicate {
+    public IsIdentityComparableWith {
+      expectedType = expectedType.unwrappedType();
+    }
 
+    @Override
+    public TriBool test(PythonType otherMaybeWrappedType) {
```

Comment:
I'm not entirely sure that the behaviour of this method and `InferredType.isIdentityComparableWith` is the same.

My understanding of this check is that it checks if two types are the same, or, if one is a union, one of its candidates is the same as the other type. Additionally, if one of the two types is unknown, the result is always `TriBool.UNKNOWN`.